src/visitors/semantic/utils.rs

```diff
@@ -1,17 +1,36 @@
+use crate::cache;
 use crate::protocol::properties::{Location, Position, Range};
-use crate::utils::get_file_contents_from_uri;
 
 use std::rc::Rc;
 
-use flux::semantic::analyze_source;
 use flux::semantic::nodes::Package;
 use flux::semantic::walk::Node;
 
+use flux::parser::parse_string;
+use flux::semantic::analyze;
+
+pub fn analyze_source(
+    source: &str,
+) -> Result<flux::semantic::nodes::Package, String> {
+    let file = parse_string("", source);
```

Comment:
Should use this. 
```
let file = Parser::new(source).parse_file("".to_string());
```
parse_string will only parse one statement, if file has multiple statements, the rest nodes will not be parsed